### PR TITLE
Add support for AWS AssumeRole in user clusters

### LINKED ISSUE #8005: Add API support for AWS AssumeRole
To support the AWS AssumeRole functionality for user clusters, the API also needs to be able to assume the IAM role. This is necessary to validate the credentials and to fetch resources like VPCs from the correct AWS account for the wizard.

- [x] Role ARN (for role to be assumed) and external id need to be added to Cloud Provider Spec 
- [x] AWS client needs to support AssumeRole: https://github.com/kubermatic/kubermatic/blob/a842a844d72c13fee16c894af337591a596e6de2/pkg/provider/cloud/aws/client.go#L36
- [x] Add AssumeRole credentials to AWS raw config (machine controller changes need to be merged first): https://github.com/kubermatic/kubermatic/blob/master/pkg/resources/machine/common.go#L95

### PR DESCRIPTION
**Pull request by simon-wessel**
**What this PR does / why we need it**:
Add API support for the AWS AssumeRole functionality to run user clusters in external AWS accounts.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubermatic/kubermatic/issues/8005

**Special notes for your reviewer**: [Epic](https://github.com/kubermatic/kubermatic/issues/6921)

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
Added API support for assuming AWS IAM roles. This allows running user clusters in e.g. external AWS accounts. (#8005, @simon-wessel)
```

/assign @moadqassem

### CODE REVIEW

**Comment by simon-wessel** [comment]
/hold
Until machine controller release

**Review by moadqassem — APPROVED** [review]
/approve
/lgtm

File: charts/kubermatic-operator/Chart.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 apiVersion: v1
 name: kubermatic-operator
-version: 0.3.20
+version: 0.3.21
```
Comment (kron4eg) [inline charts/kubermatic-operator/Chart.yaml]: This is already used tag, unfortunately

Comment (kron4eg) [reply charts/kubermatic-operator/Chart.yaml]: I'm not sure why it's not yet detected as a git conflict, but it certainly will be.

**Comment by simon-wessel** [comment]
/hold
Found something

**Comment by simon-wessel** [comment]
/unhold
Fixed an issue with the function argument order and increased kubermatic operator helm chart version.

File: pkg/apis/kubermatic/v1/cluster.go [hunk 2]
```diff
@@ -705,9 +705,11 @@ type BringYourOwnCloudSpec struct{}
 type AWSCloudSpec struct {
 	CredentialsReference *providerconfig.GlobalSecretKeySelector `json:"credentialsReference,omitempty"`
 
-	AccessKeyID     string `json:"accessKeyID,omitempty"`
-	SecretAccessKey string `json:"secretAccessKey,omitempty"`
-	VPCID           string `json:"vpcID"`
+	AccessKeyID          string `json:"accessKeyID,omitempty"`
+	SecretAccessKey      string `json:"secretAccessKey,omitempty"`
+	AssumeRoleARN        string `json:"assumeRoleArn,omitempty"`
```
Comment (kron4eg) [inline pkg/apis/kubermatic/v1/cluster.go]:
```suggestion
	AssumeRoleARN        string `json:"assumeRoleARN,omitempty"`
```

File: pkg/apis/kubermatic/v1/preset.go [hunk 3]
```diff
@@ -306,6 +306,9 @@ type AWS struct {
 	AccessKeyID     string `json:"accessKeyID"`
 	SecretAccessKey string `json:"secretAccessKey"`
 
+	AssumeRoleARN        string `json:"assumeRoleArn,omitempty"`
```
Comment (kron4eg) [inline pkg/apis/kubermatic/v1/preset.go]:
```suggestion
	AssumeRoleARN        string `json:"assumeRoleARN,omitempty"`
```

File: pkg/crd/kubermatic/v1/preset.go [hunk 4]
```diff
@@ -314,6 +314,9 @@ type AWS struct {
 	AccessKeyID     string `json:"accessKeyId"`
 	SecretAccessKey string `json:"secretAccessKey"`
 
+	AssumeRoleARN        string `json:"assumeRoleArn"`
+	AssumeRoleExternalID string `json:"assumeRoleExternalId,omitempty"`
```
Comment (kron4eg) [inline pkg/crd/kubermatic/v1/preset.go]:
```suggestion
	AssumeRoleARN        string `json:"assumeRoleARN"`
	AssumeRoleExternalID string `json:"assumeRoleExternalID,omitempty"`
```

**Review by kron4eg — COMMENTED** [review]
you've missed the cloners part for `pkg/apis/kubermatic/v1/AWS` type

**Comment by maciaszczykm** [comment]
/approve

From the API side it looks good. I believe it will not be approved before @kron4eg comments will be solved :)

**Review by kron4eg — APPROVED** [review]
/approve

**Pull request merged**